catalog/content/core/content-core-videothumbnailplugin/src/main/java/ddf/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -0,0 +1,242 @@
+                        ffmpegPath);
+            }
+        }
+    }
+
+    @Override
+    public CreateResponse process(final CreateResponse input) throws PluginExecutionException {
+        // TODO: How to handle application/octet-stream?
+        if (!input.getCreatedContentItem()
+                .getMimeType()
+                .getPrimaryType()
+                .equalsIgnoreCase("video")) {
+            return input;
+        }
+
+        final CreateRequest request = input.getRequest();
+        final Map<String, Serializable> properties = request.getProperties();
+        final ContentItem contentItem = input.getCreatedContentItem();
+
+        createThumbnail(contentItem, properties);
+
+        return new CreateResponseImpl(request,
+                contentItem,
+                input.getResponseProperties(),
+                properties);
+    }
+
+    @Override
+    public UpdateResponse process(final UpdateResponse input) throws PluginExecutionException {
+        // TODO: How to handle application/octet-stream?
+        if (!input.getUpdatedContentItem()
+                .getMimeType()
+                .getPrimaryType()
+                .equalsIgnoreCase("video")) {
+            return input;
+        }
+
+        final UpdateRequest request = input.getRequest();
+        final Map<String, Serializable> properties = request.getProperties();
+        final ContentItem contentItem = input.getUpdatedContentItem();
+
+        createThumbnail(contentItem, properties);
+
+        return new UpdateResponseImpl(request,
+                contentItem,
+                input.getResponseProperties(),
+                properties);
+    }
+
+    private CommandLine getFFmpegCommand(final String videoFilePath,
+            final String thumbnailFilePath) {
+        final String suppressPrintingBannerFlag = "-hide_banner";
+        final String inputFileFlag = "-i";
+        final String filterChainFlag = "-vf";
+        final String filterChain = "thumbnail,scale=200:-1";
+        final String videoFramesToOutputFlag = "-frames:v";
+        final String videoFramesToOutput = "1";
+        final String overwriteExistingFileFlag = "-y";
+
+        return new CommandLine(ffmpegPath).addArgument(suppressPrintingBannerFlag)
+                .addArgument(inputFileFlag)
+                .addArgument(videoFilePath, false /* Don't handle quoting */)
+                .addArgument(filterChainFlag)
+                .addArgument(filterChain)
+                .addArgument(videoFramesToOutputFlag)
+                .addArgument(videoFramesToOutput)
+                .addArgument(thumbnailFilePath, false /* Don't handle quoting */)
+                .addArgument(overwriteExistingFileFlag);
+    }
+
+    private DefaultExecuteResultHandler executeFFmpeg(final String videoFilePath,
+            final String thumbnailFilePath) throws IOException {
+        final ExecuteWatchdog watchdog = new ExecuteWatchdog(15 * 1000);
```

Comment:
I assume it is unlikely for FFmpeg to run longer than 15 seconds even for large video clips.